src/details/ArborX_DetailsUtils.hpp

```diff
@@ -585,6 +586,41 @@ template <typename View>
   return clone(ExecutionSpace{}, v);
 }
 
+template <typename ExecutionSpace, typename View>
+Kokkos::View<int *, typename View::memory_space>
+computeOffsetsInOrderedView(ExecutionSpace const &exec_space, View view)
```

Comment:
Did you consider passing the offset as argument?  It would enable setting the view label at call site and chose the integer type.